### DISCUSSION TITLE
🐛 biome_deserialize does not process escapes

### DISCUSSION THREAD

**Discussion by NicholasLYang** [biomejs/biome#2529] [Ideas]
### Environment information

```block
Not really relevant but here:

CLI:
  Version:                      1.5.2
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.5.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0
```

### What happened?

1. Deserialize `"\\"` using `biome_deserialize`
2. Note that it returns `\\`, not `\`

[playground link](https://biomejs.dev/playground/?files.main.json=IgBcAFwAIgA%3D) (although I'm aware that this only shows that the parser doesn't process escapes)
[repro here](https://github.com/NicholasLYang/biome-deserialize-repro)

### Expected result

Escapes should be processed

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
I think this a good feature, not a bug. PRs are welcome!

**Reply by NicholasLYang** [reply]
Just confirming, you're saying the current behavior is a feature or adding escape processing is a feature? I can open a PR if it's the second option.
